core/arch/arm/plat-sam/matrix.c

```diff
@@ -507,6 +54,30 @@ static unsigned int matrix_read(int base, unsigned int offset)
 	return io_read32(offset + base);
 }
 
+vaddr_t matrix_base(unsigned int matrix)
+{
+	unsigned int i = 0;
+	struct matrix *pmatrix = NULL;
+
+	do {
+		pmatrix = matrix_get(i++);
+		if (pmatrix)
+			if (pmatrix->matrix == matrix)
+				break;
+	} while (pmatrix);
+	if (pmatrix) {
+		if (cpu_mmu_enabled()) {
+			if (!pmatrix->va)
+				pmatrix->va = phys_to_virt(pmatrix->base_addr,
+							   MEM_AREA_IO_SEC, 1);
```

Comment:
The last argument to `phys_to_virt()` is the length of the of area, `1` is a valid argument, but it's more secure to supply the actual size that is going to be used.

By the way, this looks similar to what `io_pa_or_va_secure()` does.